client/src/components/Tea/TeaEditor/TeaEditorContainer.jsx

```diff
@@ -122,6 +122,22 @@ export class TeaEditorContainer extends React.Component {
     this.props.getTeaList(this.props.userID);
   }
 
+  componentDidUpdate(prevProps) {
+    // Typical usage (don't forget to compare props):
+    if (this.props.currentTea.id !== prevProps.currentTea.id) {
+      this.setState({
+        ...this.state,
```

Comment:
FYI, it's not necessary to spread the existing state like this; you can just pass an object containing the key-values you wish to update.  `setState` will perform a shallow merge of the existing state and the object you pass it: https://reactjs.org/docs/react-component.html#setstate